pilot/pkg/networking/core/listener_waypoint.go

```diff
@@ -363,6 +363,16 @@ func (lb *ListenerBuilder) buildWaypointInternal(wls []model.WorkloadInfo, svcs
 			}
 		}
 	}
+	tlsInspectorNeeded := func() bool {
```

Comment:
We need to do the `FilterDisabled: listenerPredicateExcludePorts` thing like with HTTP inspector. Else we will have a regression from the fix that supplied

Reply:
I was testing this change manually for HTTP xor TLS service, not a service that had both these ports, so I didn't notice this regression. Thanks for vigilance.